sscanss/app/dialogs/misc.py

```diff
@@ -1373,6 +1377,186 @@ def plot(self):
+        :param command_id: Value of the enum describing which command has been sent
+        :type command_id: int
+        """
+        commands = [CommandID.MovePositioner, CommandID.ChangePositionerBase, CommandID.ChangePositioningStack]
+        if command_id in commands:  # Positioning stack moved, changed, or updated
+            self.setFiducialsData()
+            self.setMatrixData()
+
+    def setFiducialsData(self):
+        """Sets the table header and inserts the data values into the cells"""
+        self.fiducial_table_widget.clear()
+        self.fiducial_table_widget.setHorizontalHeaderLabels(['X (mm)', 'Y (mm)', 'Z (mm)'])
+        self.fiducial_table_widget.setRowCount(len(self.parent.presenter.model.fiducials['points']))
+        self.fiducial_table_widget.horizontalHeader().setSectionResizeMode(QtWidgets.QHeaderView.Stretch)
+
+        self.fiducials_coordinates = self.fiducialToPosition()
+
+        for row, entry in enumerate(self.fiducials_coordinates):
+            for column in range(3):
+                fiducial_value = "{:.4f}".format(entry[column])
+                item = QtWidgets.QTableWidgetItem(fiducial_value)
+                item.setTextAlignment(QtCore.Qt.AlignCenter)
+                self.fiducial_table_widget.setItem(row, column, item)
+
+    def closeEvent(self, event):
+        self.matrix_table_widget.clear()
+        self.fiducial_table_widget.clear()
+        event.accept()
+
+    def showEvent(self, event):
+        self.setFiducialsData()
+        self.setMatrixData()
+        event.accept()
+
+    def fiducialToPosition(self):
+        """Converts fiducial positions from sample coordinate frame to instrument frame
+        :return: Fiducial coordinates in instrument frame
+        :rtype: np.ndarray
+        """
+        base_data = self.parent.presenter.model.fiducials
+        if self.parent.presenter.model.alignment is None:
+            self.banner.showMessage(
+                'Sample has not been aligned on instrument, coordinates are in sample coordinate frame.',
+                MessageType.Warning)
+            return base_data['points']
+
+        stack = self.parent.presenter.model.instrument.positioning_stack
+        pose = stack.fkine(stack.set_points, set_point=False) @ stack.tool_link
+        transform = pose @ self.parent.presenter.model.alignment
+        _matrix = transform[0:3, 0:3].transpose()
+        offset = transform[0:3, 3].transpose()
+
+        return base_data.points @ _matrix + offset
+
+    def exportFiducialsPoints(self):
+        """Writes out the data to a .fpos file if the sample has been aligned on the instrument"""
+        if self.parent.presenter.model.alignment is None:
+            self.banner.showMessage('Sample has not been aligned on instrument.', MessageType.Warning)
+            return
+
+        name, _ = QtWidgets.QFileDialog.getSaveFileName(self, 'Save Fudicials', '', "fiducial file (*.fpos)")
```

Comment:
Please use the `FileDialog` in `sscanss.core.util.widgets.py`, this ensures the appropriate extension is always appended